lib/luanetfilter.c

```diff
@@ -119,17 +131,20 @@ static int luanetfilter_register(lua_State *L)
 	nfops->hook = luanetfilter_hook;
 	nfops->dev = NULL;
 	nfops->priv = nf;
-	lunatik_setinteger(L, 1, nfops, pf);
-	lunatik_setinteger(L, 1, nfops, hooknum);
-	lunatik_setinteger(L, 1, nfops, priority);
+	luanetfilter_setinteger(L, 1, nfops, pf);
+	luanetfilter_setinteger(L, 1, nfops, hooknum);
+	luanetfilter_setinteger(L, 1, nfops, priority);
+
+	lua_getfield(L, 1, "mark");
+	nf->mark = lua_tointeger(L, -1);
+	lua_pop(L, 1); /* mark */
```

Comment:
can we replace this with `luanetfilter_setinteger(L, 1, nf, mark)` ??

Reply:
I thought the same, but `luanetfilter_setinteger` uses `lunatik_checkfield`, which will fail if `mark = nil`, the plan is to only care about `mark` if it's set and different than 0.. perhaps we can create another macro, but I found it a bit overkill for just this case.. if you think it's worth it, I can add..

Reply:
I don't think we have more use cases for it as of now.

Reply:
We can add it later